src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,425 @@
+        return block_size
+
+    def set_device_faulty(self):
+        """
+        Set the devices to Faulty one by one and wait for rebuild to complete.
+
+        Returns:
+            None
+        """
+        #Get the device ids from all servers and try to eject the disks
+        device_ids = get_device_ids(self.dmg, self.hostlist_servers)
+
+        #no_of_servers and no_of_drives can be set from test yaml.
+        #1 Server, 1 Drive = Remove single drive from single server
+        for num in range(0, self.no_of_servers):
+            server = self.hostlist_servers[num]
+            for disk_id in range(0, self.no_of_drives):
+                self.dmg.hostlist = server
+                self.dmg.storage_set_faulty(device_ids[server][disk_id])
+
+                # Wait for rebuild to start
+                self.pool.wait_for_rebuild(True)
+
+                # Wait for rebuild to complete
+                self.pool.wait_for_rebuild(False)
+
+    def start_ior_load(self):
+        """
+        Method to Fill up the server. It will get the maximum Storage space and
+        create the pool.Fill up the server based on % amount given using IOR.
+
+        Returns:
+            None
+        """
+        #Method to get the storage capacity.Replace with dmg options in future
+        #when it's available. This is time consuming so store the size in
+        #file to avoid rerunning the same logic for future test cases.
+        #This will be only run for first test case.
+
+        avocao_tmp_dir = os.environ['AVOCADO_TESTS_COMMON_TMPDIR']
+        capacity_file = os.path.join(avocao_tmp_dir, 'storage_capacity')
+        if not os.path.exists(capacity_file):
+            #Stop server but do not reset.
+            self.stop_servers()
+            total_nvme_capacity = self.get_server_capacity()
+            with open(capacity_file,
+                      'w') as _file: _file.write('{}'
+                                                 .format(total_nvme_capacity))
+            #Start the server.
+            self.start_servers()
+        else:
+            total_nvme_capacity = open(capacity_file).readline().rstrip()
+
+        print("Server Stoarge capacity = {}".format(total_nvme_capacity))
+        # Create a pool
+        self.pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+        self.pool.get_params(self)
+        self.pool.nvme_size.update('{}'.format(total_nvme_capacity))
+        self.pool.create()
+        print("Pool Usage Percentage - Before - {}"
+              .format(self.pool.pool_percentage_used()))
+
+        # Create the IOR threads
+        job = threading.Thread(target=self.start_ior_thread,
+                               kwargs={"results":self.out_queue,
+                                       "operation": 'Write'})
+        # Launch the IOR thread
+        job.start()
+
+        #Set NVMe device faulty if it's set
+        if self.set_faulty_device:
+            time.sleep(60)
```

Comment:
Any idea why this timeout is required?

Reply:
This is timeout to wait to generate faulty device after IOR start so IOR will have written some data when disk failure happens.